keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/constraint/ImageExistsConstraintEvaluator.kt

```diff
@@ -44,37 +41,21 @@ class ImageExistsConstraintEvaluator(
     version: String,
     deliveryConfig: DeliveryConfig,
     targetEnvironment: Environment
-  ): Boolean {
-    if (artifact !is DebianArtifact) {
-      return true
-    }
-
-    val image = findMatchingImage(version, artifact.vmOptions)
-    return image != null
-  }
-
-  private data class Key(
-    val account: String,
-    val version: String,
-    val regions: Set<String>
-  )
+  ): Boolean =
+    artifact !is DebianArtifact || imagesExistInAllRegions(version, artifact.vmOptions)
 
-  private val cache = cacheFactory
-    .asyncLoadingCache<Key, NamedImage>(cacheName = "namedImages") { key ->
-      log.debug("Searching for baked image for {} in {}", key.version, key.regions.joinToString())
-      imageService.getLatestNamedImageWithAllRegionsForAppVersion(
-        // TODO: Frigga and Rocket version parsing are not aligned. We should consolidate.
-        appVersion = AppVersion.parseName(key.version)
-          ?: throw SystemException("Invalid AMI app version: ${key.version}"),
-        account = key.account,
-        regions = key.regions
+  private fun imagesExistInAllRegions(version: String, vmOptions: VirtualMachineOptions): Boolean =
+    runBlocking {
+      imageService.getLatestNamedImageForAppVersionInRegions(
```

Comment:
nit, and I know this wasn't changed in this PR, but I find this function name unnecessarily long/complicated. We're naming the function after the parameters it takes... It could just be `getLatestNamedImage` with the right params.

Reply:
I changed it as suggested